src/col/vct/col/typerules/CoercingRewriter.scala

```diff
@@ -1670,8 +1673,22 @@ abstract class CoercingRewriter[Pre <: Generation]() extends AbstractRewriter[Pr
       case w @ WandPackage(expr, stat) => WandPackage(res(expr), stat)(w.blame)
       case VeyMontAssignExpression(t,a) => VeyMontAssignExpression(t,a)
       case CommunicateX(r,s,t,a) => CommunicateX(r,s,t,a)
-      case PVLCommunicate(s, r) => PVLCommunicate(s, r)
-      case Communicate(r, s) => Communicate(r, s)
+      case PVLCommunicate(s, r) if r.fieldType == s.fieldType => PVLCommunicate(s, r)
```

Comment:
@pieter-bos I think we could support coercions relatively easily, but I'd like to postpone that feature for now. Is this the right way of encoding that the types must be exactly the same? Also, two practical problems: 1) the error is mega ugly, it prints the entire program on error, and 2) this error is structured the same way as for the assign statement, but it results in a crash instead of an error code in the test suite. Maybe we can discuss this on tuesday :)

Reply:
Maybe have a look at `coerceYields` (although I really want to get rid of that at some point)

Reply:
Okay, will do. I'm assuming you're otherwise okay with how I handled the type equality.